exporters/prometheus/src/exporter_utils.cc

```diff
@@ -55,7 +51,11 @@ std::vector<prometheus_client::MetricFamily> PrometheusExporterUtils::TranslateT
               nostd::get<sdk::metrics::SumPointData>(point_data_attr.point_data).is_monotonic_;
         }
         const prometheus_client::MetricType type = TranslateType(kind, is_monotonic);
-        metric_family.type                       = type;
+        prometheus_client::MetricFamily metric_family;
+        metric_family.type = type;
+        metric_family.name = MapToPrometheusName(metric_data.instrument_descriptor.name_,
+                                                 metric_data.instrument_descriptor.unit_, type);
+        metric_family.help = metric_data.instrument_descriptor.description_;
```

Comment:
This code moved inside the loop on data points.

The metric name, type, help, etc should be constant for all data points, so can this be moved outside ?

Even if `type` is needed and not available elsewhere, peek at the first data point before entering the loop on data points ?

Reply:
+1.  As long as it is safe to assume all data points have the same type, peeking at the first seems like the right thing to do.  I took that approach in https://github.com/open-telemetry/opentelemetry-cpp/pull/2288/files#diff-e92aebc994eed5c134b09ecde972d9db39a9bfc46edc6d93b0b5a4c98e4ff27fR46.

Reply:
thanks, moved back to where it was.